models/cartDao.js

```diff
@@ -0,0 +1,49 @@
+const appDataSource = require('./appDataSource');
+
+const addToCart = async (productId, userId) => {
+  return await appDataSource.query(
+    `INSERT INTO carts(
+            product_id,
+            user_id,
+            quantity
+        ) VALUES (?, ?, 1)`,
+    [productId, userId]
+  );
+};
+
+const deleteFromCart = async (pId, userId) => {
+  return await appDataSource.query(
+    `DELETE FROM carts
+    WHERE user_id = ? AND product_id= ?`,
+    [userId, pId]
+  );
+};
```

Comment:
controller 코드를 보니 pId 라고 받는 product id값을 쿼리로 받고 있는 것 같습니다! 이는 REST 아키텍쳐에서 삭제 기능을 구현할 때 활용하기 좋은 방법이라고 생각됩니다.

query string으로 매개변수를 받게 되면 다음과 같이 request uri가 작성됩니다.
```shell
// 1개를 보내는 경우
[DELETE] http://127.0.0.1:3000/carts?product-id=1

// 여러개를 보내는 경우
[DELETE] http://127.0.0.1:3000/carts?product_id=1&product_id=2&product_id=3
```

위와 아래를 각각 콘솔로그로 출력시 다음과 같은 값이 나옵니다.
```javascript
const {product_id: productId} = req.query;

console.log(productId)
// case 1  ->   1
// case 2  ->  ['1', '2', '3']
```
해당 productId 값을 활용하면 아래와 같이 query에서도 매우 간단하게 처리 할 수 있습니다.
```suggestion
const deleteFromCart = async (pId, userId) => {
  return await appDataSource.query(
    `DELETE FROM carts
    WHERE user_id = ? AND product_id IN (?)`,
    [userId, pId]
  );
};
```
IN 을 쓰게 되면 productId로 하나가 넘어오던, 여러개가 배열로 넘어오던 삭제 처리를 일괄적으로 할 수 있고, 이는 당장 우리서비스가 클라이언트에서 선택 삭제 혹은 전체 삭제를 지원하지 않다가 추후에 이러한 기능이 지원되더라도 API 수정은 없거나 최소화 할 수 있습니다.